FerramAerospaceResearch.Base/UnityJobs/OcclusionJobs.cs

```diff
@@ -0,0 +1,381 @@
+            float3 origin = float3.zero + math.length(extents) * normal;
+            float2 min = new float2(float.PositiveInfinity, float.PositiveInfinity);
+            float2 max = new float2(float.NegativeInfinity, float.NegativeInfinity);
+            float3 ext = extents / 2;
+            float3 center = boundsCenter;
+            unsafe
+            {
+                float3* arr = stackalloc[]
+                {
+                    center - ext,
+                    new float3(center.x - ext.x, center.y - ext.y, center.z + ext.z),
+                    new float3(center.x - ext.x, center.y + ext.y, center.z - ext.z),
+                    new float3(center.x - ext.x, center.y + ext.y, center.z + ext.z),
+                    new float3(center.x + ext.x, center.y - ext.y, center.z - ext.z),
+                    new float3(center.x + ext.x, center.y - ext.y, center.z + ext.z),
+                    new float3(center.x + ext.x, center.y + ext.y, center.z - ext.z),
+                    center + ext,
+                };
+
+                for (int i = 0; i < 8; i++)
+                {
+                    // Project the bounding box onto the plane
+                    float dist = math.dot(arr[i] - origin, normal);
+                    float3 projection = arr[i] - dist * normal;
+                    // Rotate the projection back towards the vessel alignment
+                    projection = math.mul(math.inverse(quaternions[index]), projection);
+
+                    // The x,y coords of projection are again aligned to the vessel axes.
+                    min.x = math.min(min.x, projection.x);
+                    min.y = math.min(min.y, projection.y);
+                    max.x = math.max(max.x, projection.x);
+                    max.y = math.max(max.y, projection.y);
+                }
+            }
+
+            bounds[index] = new float2(math.abs(max.x - min.x), math.abs(max.y - min.y));
+        }
+    }
+
+    [BurstCompile]
+    public struct SetIntervalAndDimensionsJob : IJobParallelFor
+    {
+        [ReadOnly] public NativeArray<float2> bounds;
+        [ReadOnly] public int maxDim;
+        [ReadOnly] public float resolution;
+        [WriteOnly] public NativeArray<float2> interval;
+        [WriteOnly] public NativeArray<int2> dims;
+
+        public void Execute(int index)
+        {
+            int minXsize = (int)math.ceil(bounds[index].x / resolution);
+            int minYsize = (int)math.ceil(bounds[index].y / resolution);
+            dims[index] = new int2(math.min(maxDim, minXsize), math.min(maxDim, minYsize));
+
+            interval[index] = new float2(math.max(bounds[index].x / maxDim, resolution), math.max(bounds[index].y / maxDim, resolution));
+        }
+    }
+
+    [BurstCompile]
+    public struct SetVectorsJob : IJobParallelFor
+    {
+        [ReadOnly] public NativeArray<quaternion> quaternions;
+        [ReadOnly] public float4x4 localToWorldMatrix;
```

Comment:
`RigidTransform` may be more appropriate than `float4x4` for translation + rotation

Reply:
Not sure what to do here.  Maybe, but the localToWorldMatrix is easily passed from outside.  Maybe I'll get an improvement, but SetVectorsJob [now as modified] consumes a very small fraction of the compute here.  (Current run on my PC -- maybe not the most useful reference without a lot more context -- is 0.028ms)